tests/integration/helpers.ts

```diff
@@ -23,8 +23,12 @@ export interface IntegrationTest {
 export function setupIntegrationTest(getUserConfig: () => UserConfig): IntegrationTest {
     let mcpClient: Client | undefined;
     let mcpServer: Server | undefined;
+    let oldDoNotTrackValue: string | undefined;
 
     beforeAll(async () => {
+        // GET DO_NOT_TRACK value
+        oldDoNotTrackValue = process.env.DO_NOT_TRACK;
+        process.env.DO_NOT_TRACK = "1";
         const userConfig = getUserConfig();
```

Comment:
instead of keep track of DO_NOT_TRACK, why not
```suggestion
        const userConfig = {
           ...getUserConfig(),
           telemetry: "disabled"
       };
```

Reply:
actually, let me try to sort our default test config everywhere

Reply:
this won't work too cause telemetry imports config so it's not injected, only env variables would work or we can refactor

Reply:
I'll refactor a bit